test/test_script.py

```diff
@@ -943,3 +950,31 @@ def test_cmd_slice(self):
             self.parser.eval("$slice(abc; def,0)")
         with self.assertRaisesRegex(ScriptError, areg):
             self.parser.eval("$slice(abc; def),0,1,:,extra")
+
+    def test_cmd_datetime(self):
+        # Save origninal datetime object and substitute one returning
+        # a fixed now() value for testing.
+        original_datetime = datetime.datetime
+        datetime.datetime = _DateTime
+
+        context = Metadata()
+        context["foo"] = "%Y%m%d%H%M%S.%f"
```

Comment:
If your intent is to restore original `datetime.datetime` whatever happens, those two lines should also be in the try/finally block.

Reply:
Good point.  I was thinking to only include the lines doing the testing, but I see that an exception thrown here would cause it not to be restored.